doc/devices/ahs_local.rst

```diff
@@ -0,0 +1,133 @@
+    functions should follow the conventions specified in the PennyLane docs to ensure correct unit conversion.
+    See `rydberg_interaction <https://docs.pennylane.ai/en/stable/code/api/pennylane.pulse.rydberg_interaction.html>`_
+    and `rydberg_drive <https://docs.pennylane.ai/en/stable/code/api/pennylane.pulse.rydberg_drive.html>`_ in
+    Pennylane for specification of expected input units, and examples for creating hardware-compatible
+    `ParametrizedEvolution <https://docs.pennylane.ai/en/stable/code/api/pennylane.pulse.ParametrizedEvolution.html>`_
+    operators in PennyLane.
+
+Creating a register
+^^^^^^^^^^^^^^^^^^^
+
+The atom register defines where the atoms will be located, which determines the strength of the interaction
+between the atoms. Here we define coordinates for the atoms to be placed at (in micrometers), and create a constant
+interaction term for the Hamiltonian:
+
+.. code-block:: python
+
+    # number of coordinate pairs must match number of device wires
+    coordinates = [[0, 0], [0, 5]]  
+
+    H_interaction = qml.pulse.rydberg_interaction(coordinates)
+
+Creating a drive
+^^^^^^^^^^^^^^^^^^^^^^^
+
+We can create a drive with a global component and (positive) local detunings. If the local detunings are time-dependent,
+they must all have the same time-dependent envelope, but can have different, positive scaling factors.
+
+.. code-block:: python
+
+    # gaussian amplitude function (qml.pulse.rect enforces 0 at start and end for hardware)
+    def amp_fn(p, t):
+        f = p[0] * jnp.exp(-(t - p[1])**2 / (2 * p[2]**2))
+        return qml.pulse.rect(f, windows=[0.1, 1.7])(p, t)
+
+    # defining a linear detuning
+    def det_fn_global(p, t):
+        return p * t
+
+    def det_fn_local(p, t):
+        return p * t**2
+
+    # creating a global drive on all wires
+    H_global = qml.pulse.rydberg_drive(amplitude=amp_fn, phase=0, detuning=det_fn_global, wires=[0, 1])
+
+    # creating local drives
+    # note only local detuning is currently supported, so amplitude and phase are set to 0
+    H_local0 = qml.pulse.rydberg_drive(amplitude=0, phase=0, detuning = det_fn_local, wires=[0])
+    H_local1 = qml.pulse.rydberg_drive(amplitude=0, phase=0, detuning = det_fn_local, wires=[1])
+
+    # full hamiltonian
+    H = H_interaction + H_global + H_local0 + H_local1
+
+
+Executing an AHS program
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+    @qml.qnode(device_local)
+    def circuit(params):
+        qml.evolve(H)(params, t=1.5)
+        return qml.sample()
+
+    # amp_fn expects p to contain 3 parameters
+    amp_params = [2.5, 1, 0.3]
+    # global_det_fn expects p to be a single parameter
+    det_global_params = 0.2
+    # each of the local drives take a single parameter for p
+    # the detunings have the same shape, but vary by scaling factor p
+    local_params1 = 0.5
+    local_params2 = 1
```

Comment:
Can you mention in code comments or text which parameters will be multiplied by `2 pi 1e6`? Its actually not clear to myself right now

Reply:
These parameters are not multiplied by anything - they are input for the functions that define amplitude and detunings. The amplitudes and detunings output by these functions are converted from MHz to rad/s.

I don't think adding information about this here would be clarifying. I think we have a good plan for documentation and examples on the PennyLane side in the helper functions and demos, and we should leave this as is.